### ISSUE TITLE
Lumos: dependencies or peerDependencies

### ISSUE THREAD

**Issue by ShookLyngs** [sporeprotocol/spore-sdk#64]
Lumos is a collection of packages that developers commonly use for developing dapps on CKB. It serves as a foundation for the spore-sdk and plays a significant role in managing cells and witnesses.

However, I sometimes encounter a dilemma. The spore-sdk lists lumos as a dependency in its `package.json`, but when lumos releases a new version, the spore-sdk doesn't automatically update to the latest version. It lags behind in this regard.

One potential solution is to move lumos from dependencies to peerDependencies. This allows developers to freely choose which version of lumos they want to use. However, this solution has a drawback: unlike dependencies, developers must manually install peerDependencies into their dapp projects.

In conclusion:
- If specify in `dependencies`:
  - Pro - Developers don't have to manually manage lumos in most cases
  - Con - When lumos has a breaking update/fix, the spore-sdk must also release a new version
- If specify in `peerDependencies`:
  - Pro - Developers can freely choose which version of lumos they want to use
  - Con - Developers must manually specify and install lumos in their dapp projects

**Comment by homura** [comment]
It depends on how users utilize the Spore SDK. 

If we only require Spore SDK methods and not other Lumos methods, then installing the Spore SDK is better. 

However, if we need other CKB-related methods in most situations when using Spore SDK, then it's more flexible to use `peerDependencies` with a ranged version. This approach would be beneficial since we only need one Lumos in the build to create a lighter production build.

BTW, Lumos has [refactored](https://lumos-website.vercel.app/migrations/migrate-to-v0.22) the entry package `@ckb-lumos/lumos` from 0.22, and we don't need to import several child packages in most cases. This will make it simpler when Spore SDK chooses the `peerDependencies` solution

https://github.com/sporeprotocol/spore-sdk/blob/6c5734157cc5efd96faf610b731e35d22afab06c/packages/core/package.json#L20-L26

**Comment by ashuralyk** [comment]
@ShookLyngs this issue gets a potential resolution for a better experience?

**Comment by ShookLyngs** [comment]
> this issue gets a potential resolution for a better experience?

For now, I suggest we keep the lumos family in the `dependencies`. Moving them to `peerDependencies` can potentially introduce other unexpected issues for developers (too many peer dependencies), which is not our intention.

Regarding the improvement resolution, as homura mentioned, if we can reduce the lumos-related dependency count in the spore-sdk, such as only having `@ckb-lumos/lumos` as a dependency, we can consider moving it to the `peerDependencies`. This change is unlikely to cause troubles for developers.
